src/mono/wasm/Wasm.Build.Tests/WasmSdkBasedProjectProvider.cs

```diff
@@ -105,7 +105,10 @@ public void AssertBundle(AssertWasmSdkBundleOptions assertOptions)
             _ => throw new ArgumentOutOfRangeException(nameof(assertOptions.ExpectedFileType))
         };
         string buildType = assertOptions.IsPublish ? "publish" : "build";
-        foreach (string nativeFilename in new[] { "dotnet.native.wasm", "dotnet.native.js" })
+        var nativeFilesToCheck = assertOptions.RuntimeType == RuntimeVariant.MultiThreaded ?
```

Comment:
Could we assert there is exactly one file in the collections that are supposed to contain one file?
It probably should happen in the base class

Reply:
You mean that there are not 2 files with same names in the same path? I thought about it but technically it's impossible to have 2 files of same name at same location, so checking for it should not be needed.

Reply:
I would like to check content of `blazor.boot.json`

Reply:
That's already done in https://github.com/dotnet/runtime/blob/eeadd653e1982d7037a93a9ab38129c07336e7db/src/mono/wasm/Wasm.Build.Tests/ProjectProviderBase.cs#L455
it compares the count of resources in `blazor.boot.json` with fixed expected resources set that contains one `dotnet.native.worker.js` for MT.